research-software-directory/RSD-as-a-service
# Allow user to delete account from RSD

# Admin section and allow user to remove account

Closes #740 

Changes proposed in this pull request:
* Created rsd admin section with the menu options on the left side
* Added rsd accounts section where rsd_admin can remove account
* Added remove account button at the user profile page. If user has not accepted terms of use we show warning which explains that user needs to accept terms of use or send an request to support for account removal.
* Color of the progress bar in the header changed to accent from primary color (better contrast)

**Question/Remark**
It seems that regular users do need to first accept terms of use before being able to remove account. We have implemented checks at db level that do not allow data manipulation without TOS accepted. The check is not applied to rsd_admin role so rsd_admin can remove account(s) without accepting terms of use.

How to test:
* `make start`: to rebuild app
* login as user, navigate to your settings (from user menu). Use remove account button. Your account will be removed and you will be logged out. Note! When user logs in next time the account will be recreated. 
* login as rsd_admin. Navigate to administration, rsd users section. You will see the list of RSD accounts. You can search by account id (exact value) or by part of user name, email or organisation. You can remove any account as rsd_admin. NOTE! This will not prevent user from being able to login, however all maintainer permissions for this account will be removed as well as all maintainer invites the user has created or claimed.
* check if remove account messages are correct and understandable

## Remove account (disabled)
![image](https://user-images.githubusercontent.com/9204081/222460451-f4a45e41-1359-4918-b4ab-389c3761b201.png)

## Remove account (enabled)
![image](https://user-images.githubusercontent.com/9204081/222460604-cabb547f-ed52-4051-bc93-65cbcd06f43c.png)

## Remove account (confirmation)
![image](https://user-images.githubusercontent.com/9204081/222460762-3adaca08-3152-42c9-b240-447341b98fae.png)

## RSD admin section

![image](https://user-images.githubusercontent.com/9204081/221867617-fee40e0d-805a-448b-90c3-285c17291a41.png)

## RSD admin section (confirmation)

![image](https://user-images.githubusercontent.com/9204081/222461378-026157fe-6ca3-4900-baab-bd3c6f1a3f4b.png)


PR Checklist:
